File: coolrmi/hu.qgears.coolrmi/src/hu/qgears/coolrmi/multiplexer/SocketMultiplexerSource.java
package hu.qgears.coolrmi.multiplexer;

import hu.qgears.coolrmi.messages.AbstractCoolRMIMessage;

/**
 * A datagram source that must be sent.
 * @author rizsi
 *
 */
public class SocketMultiplexerSource {
	private byte[] toSend;
	private AbstractCoolRMIMessage message;

	public byte[] getToSend() {
		return toSend;
	}
	public SocketMultiplexerSource(byte[] toSend, AbstractCoolRMIMessage message) {
		super();
		this.toSend = toSend;
		this.message=message;
	}
	@Override
	public String toString() {
		return "Socket multiplexer source: "+message.getName();
	}
	/**
	 * Callback when the last piece of this message has been sent through the (TCP) channel.
	 */
	public void sent() {
		message.sent();
	}
}

File: coolrmi/CoolRMI.Net/Multiplexer/SocketMultiplexerSource.cs
using CoolRMI.Net.Messages;

namespace CoolRMI.Net.Multiplexer
{
    public struct SocketMultiplexerSource
    {
        public byte[] ToSend { get; }
        private readonly AbstractCoolRMIMessage message;

        public SocketMultiplexerSource(byte[] toSend,
            AbstractCoolRMIMessage message)
        {
            ToSend = toSend;
            this.message = message;
        }

        public override string ToString()
        {
            return "Socket multiplexer source: " + message.Name;
        }

        /// <summary>
        /// Callback when the last piece of this message has been sent through the (TCP) channel.
        /// </summary>
        public void Sent() {
            message.Sent();
        }
    }
}
